test/sqllogictest/transform/aggregation_nullability.slt

```diff
@@ -193,26 +251,55 @@ EXPLAIN WITH(arity, join_impls) select t1.f1, count(t2.f1), sum(t2.f1), max(t2.f
 ----
 Explained Query:
   Return // { arity: 9 }
-    Filter (#2 >= 0) // { arity: 9 }
-      Reduce group_by=[#0] aggregates=[count(#2), sum(#2), max(#2), min(#2), count(#1), sum(#1), min(#1), max(#1)] // { arity: 9 }
-        Union // { arity: 3 }
-          Map (null) // { arity: 3 }
-            Union // { arity: 2 }
-              Negate // { arity: 2 }
-                Project (#0, #1) // { arity: 2 }
-                  Join on=(#0 = #2) type=differential // { arity: 3 }
-                    implementation
-                      %0:t1[#0] » %1[#0]UKA
-                    ArrangeBy keys=[[#0]] // { arity: 2 }
-                      Get materialize.public.t1 // { arity: 2 }
-                    ArrangeBy keys=[[#0]] // { arity: 1 }
-                      Distinct group_by=[#0] // { arity: 1 }
-                        Project (#0) // { arity: 1 }
-                          Get l0 // { arity: 2 }
-              Get materialize.public.t1 // { arity: 2 }
-          Project (#0, #1, #0) // { arity: 3 }
-            Get l0 // { arity: 2 }
+    Project (#0, #9, #10, #1, #3, #11, #12, #5, #7) // { arity: 9 }
+      Filter (#10 >= 0) // { arity: 13 }
+        Join on=(eq(#0, #2, #4, #6, #8)) type=delta // { arity: 13 }
+          implementation
+            %0 » %4[#0]UKAif » %1[#0]UKA » %2[#0]UKA » %3[#0]UKA
+            %1 » %4[#0]UKAif » %0[#0]UKA » %2[#0]UKA » %3[#0]UKA
+            %2 » %4[#0]UKAif » %0[#0]UKA » %1[#0]UKA » %3[#0]UKA
+            %3 » %4[#0]UKAif » %0[#0]UKA » %1[#0]UKA » %2[#0]UKA
+            %4 » %0[#0]UKA » %1[#0]UKA » %2[#0]UKA » %3[#0]UKA
+          ArrangeBy keys=[[#0]] // { arity: 2 }
+            Reduce group_by=[#0] aggregates=[max(#1)] // { arity: 2 }
+              Get l2 // { arity: 2 }
+          ArrangeBy keys=[[#0]] // { arity: 2 }
+            Reduce group_by=[#0] aggregates=[min(#1)] // { arity: 2 }
+              Get l2 // { arity: 2 }
+          ArrangeBy keys=[[#0]] // { arity: 2 }
+            Reduce group_by=[#0] aggregates=[min(#1)] // { arity: 2 }
+              Get l3 // { arity: 2 }
+          ArrangeBy keys=[[#0]] // { arity: 2 }
+            Reduce group_by=[#0] aggregates=[max(#1)] // { arity: 2 }
+              Get l3 // { arity: 2 }
+          ArrangeBy keys=[[#0]] // { arity: 5 }
+            Reduce group_by=[#0] aggregates=[count(#2), sum(#2), count(#1), sum(#1)] // { arity: 5 }
+              Get l1 // { arity: 3 }
```

Comment:
This is an interesting case in that it would not be a bad idea to split the hierarchical aggregates that operate on different attributes, but it seems to be still better to compute min/max on the same attribute in a single aggregation as you pointed out in [Slack](https://materializeinc.slack.com/archives/C02PPB50ZHS/p1673109203160559).